### DISCUSSION TITLE
How to draw the 3 axes with vsg?

### DISCUSSION THREAD

**Discussion by redflare2023** [vsg-dev/VulkanSceneGraph#714] [Q&A] [answered]
I'm a new bee of vsg. and I know how to draw the 3 axes with osg hud and camera like the sceenshot, but what should I do with vsg?
Thank for any advice!
![image](https://user-images.githubusercontent.com/123873517/215325541-6d976729-7c87-4023-a604-6b03bd573e48.png)

**Comment by robertosfield** [comment, marked as answer]
Arrows and axis rendering like you show is something I've been considering as an addition to [vsg::Builder.](https://github.com/vsg-dev/VulkanSceneGraph/blob/master/include/vsg/utils/Builder.h)  

Right now Builder can created cones and cylinders which is what is needed in for the axis arrows so could be used to build up each part.  To see how to use vsg::Builder have a look at the [vsgbuilder](https://github.com/vsg-dev/vsgExamples/blob/master/examples/utils/vsgbuilder/vsgbuilder.cpp) examples.

The HUD side could be implemented with with a separate View with the Camera's View and Projection matrices set up to do the HUD.  The [vsgcamera](https://github.com/vsg-dev/vsgExamples/blob/master/examples/app/vsgcameras/vsgcameras.cpp) and [vsgmultiviews](https://github.com/vsg-dev/vsgExamples/blob/master/examples/app/vsgmultiviews/vsgmultiviews.cpp) examples illustrate how this can be done.  For a hud you just want to clear the depth buffer.

To sync the rotation of the axis you'll want to modify the HUD View's Camera View matrix so that the rotate component is the same as the main Camera's view matrix.

**Reply by robertosfield** [reply]
I have to add putting all these features together in a single example would make for great addition to vsgExamples, so if someone has time and fancies figuring out how to do it all please jump in :-)

**Comment by redflare2023** [comment]
@robertosfield ,thank for your advice. 
I have read the vsgmultiviews sample. the background color of the second view is 0.2f,0,2f,0.2f,1.0f , but I want  transparent backgound in the view,so I modify the clear color alpha value from 1.0f to 0.0f,just like the following picture, but nothing happened. what did i miss?
![image](https://user-images.githubusercontent.com/123873517/215338374-ee3f05e8-03ce-4f21-a1a5-3bce92ee2a17.png)
by the way, there may be some error in separateRenderGraph mode. if i use separateRenderGraph mode ,the main view have no any model. when i resizing the window to some specical size(maybe ) , we can see that there are a teapot in main view. 

![image](https://user-images.githubusercontent.com/123873517/215338641-5f1555aa-e16c-48ee-a3b1-707fe096aa7c.png)
 
![image](https://user-images.githubusercontent.com/123873517/215338682-5cd0fc85-898e-4300-b8d3-f938ebb2fe6d.png)

**Reply by robertosfield** [reply]
To avoid the clear of the colour buffer you just need to change the ClearAttachements::Attachements{..} so it only contains the depth_attachment i.e.

```
VkClearValue depthClearValue{};
depthClearValue.depthStencil = {0.0f, 0};
VkClearAttachment depth_attachment{VK_IMAGE_ASPECT_DEPTH_BIT, 1, depthClearValue};

VkClearRect rect{secondary_camera->getRenderArea(), 0, 1};
auto clearAttachments = vsg::ClearAttachments::create(vsg::ClearAttachments::Attachments{depth_attachment}, vsg::ClearAttachments::Rects{rect});
renderGraph->addChild(clearAttachments);
```

**Comment by redflare2023** [comment]
ok, let me try it. thanks a lot! once i finish the code , i will commit it as a sample.
